api/src/main/java/io/strimzi/api/kafka/model/status/ListenerStatus.java

```diff
@@ -28,25 +29,37 @@
         builderPackage = Constants.FABRIC8_KUBERNETES_API
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({ "type", "addresses", "bootstrapServers", "certificates" })
+@JsonPropertyOrder({ "type", "name", "addresses", "bootstrapServers", "certificates" })
 @EqualsAndHashCode
 public class ListenerStatus implements UnknownPropertyPreserving, Serializable {
     private static final long serialVersionUID = 1L;
 
-    private String type;
+    private String name;
     private List<ListenerAddress> addresses;
     private String bootstrapServers;
     private List<String> certificates;
     private Map<String, Object> additionalProperties;
 
-    @Description("The type of the listener. " +
-            "Can be one of the following three types: `plain`, `tls`, and `external`.")
+    @Description("The type of the listener.")
+    @DeprecatedProperty(movedToPath = "name")
+    @Deprecated
     public String getType() {
-        return type;
+        return name;
     }
 
-    public void setType(String type) {
-        this.type = type;
+    @DeprecatedProperty(description = "The `type` field  setter in `ListenerStatus` has been deprecated")
```

Comment:
I don't think this is used anywhere.

Reply:
So we don't need a Deprecated tag over it ?

Reply:
Well, I'm talking about the `@DeprecatedProperty` => do you see it used for anything? It does not seem to. I think it would need to be on the getter and not on the setter.

Reply:
Done